packages/sit-onyx/src/components/OnyxFlyoutMenu/future/OnyxFlyoutMenu.vue

```diff
@@ -0,0 +1,123 @@
+<script setup lang="ts" generic="TValue extends SelectOptionValue = SelectOptionValue">
+import { createMenuButton } from "@sit-onyx/headless";
+import type { SelectOptionValue } from "../../../types";
+import { computed, ref, type VNode } from "vue";
+
+const slots = defineSlots<{
+  /**
+   * The trigger for the flyout menu
+   */
+  default(): VNode[];
+  /**
+   * OnyxListItems to show
+   */
+  options?(): VNode[];
+  /**
+   * Optional header content to display above the options.
+   */
+  header?(): unknown;
+  /**
+   * Optional footer content to display below the options (will replace `message` property).
+   */
+  footer?(): unknown;
+}>();
+
+const activeItem = ref<string>();
+
+const {
+  elements: { button, menu, menuItems, listItem, parentComponent },
+  state: { isExpanded },
+} = createMenuButton({
+  onSelect: (value) => {
+    activeItem.value = value;
+  },
+});
+
+const getSlotComponents = (vnodes: VNode[]): VNode[] => {
+  // if the slot only contains a v-for, we need to use the children here which are the "actual" slot content
+  const isVFor = vnodes.length === 1 && vnodes[0].type.toString() === "Symbol(v-fgt)";
+
+  const allNodes =
+    isVFor && Array.isArray(vnodes[0].children)
+      ? (vnodes[0].children as Extract<(typeof vnodes)[number]["children"], []>)
+      : vnodes;
+
+  return allNodes;
+};
+
+const options = computed(() => {
+  return getSlotComponents(slots.options?.() ?? []);
+});
+</script>
+
+<template>
+  <div class="onyx-flyout-menu" v-bind="parentComponent">
```

Comment:
The flyout also openes when hovering outside of the button

https://github.com/SchwarzIT/onyx/assets/67898185/df6800b0-a5a7-454e-9410-3914fbd6798e